saas/api/roles.py

```diff
@@ -670,8 +672,35 @@ class RoleByDescrQuerysetMixin(RoleDescriptionMixin, RoleQuerysetMixin):
 
     def get_queryset(self):
         return super(RoleByDescrQuerysetMixin, self).get_queryset().filter(
-            Q(role_description=self.role_description)
-            | Q(request_key__isnull=False))
+            role_description=self.role_description)
+
+
+class RoleListPagination(PageNumberPagination):
+
+    def paginate_queryset(self, queryset, request, view=None):
+        self.invited_count = queryset.filter(
+            grant_key__isnull=False).count()
+        self.requested_count = queryset.filter(
+            request_key__isnull=False).count()
+        qry = {}
+        role_status = request.query_params.get('role_status', '')
+        stts = role_status.split(',')
+        if 'invited' not in stts:
```

Comment:
`role_status` is an additional filter. By default all roles (active, invited, requested) should be returned by the API end-point, then adding `role_status=active` should return only the active roles, `role_status=invited` should return only the invited, `role_status=active,requested` should return active or requested, etc.

Reply:
I was thinking about this. I decided to do the reverse to not break the other portions of the app that rely on the old behavior (e.g. typeaheads)